src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +249,522 @@ otError DuaManager::Store(void)
+        case ThreadStatusTlv::kDuaReRegister:
+            mRegisterCurrentChildIndex = true;
+            mDelay.mFields.mCheckDelay = Mle::kImmediateReRegisterDelay;
+            break;
+        case ThreadStatusTlv::kDuaInvalid:
+            // fall through
+        case ThreadStatusTlv::kDuaDuplicate:
+            SendAddressNotification(target, static_cast<ThreadStatusTlv::DuaStatus>(status), *child);
+            IgnoreError(child->RemoveIp6Address(target));
+            break;
+        case ThreadStatusTlv::kDuaNoResources:
+            // fall through
+        case ThreadStatusTlv::kDuaNotPrimary:
+            // fall through
+        case ThreadStatusTlv::kDuaGeneralFailure:
+            mChildDuaRegisterMask.Clear(mChildIndexDuaRegistering);
+            UpdateReregistrationDelay();
+
+            break;
+        }
+    }
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+
+exit:
+    ScheduleTimer();
+}
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+void DuaManager::SendAddressNotification(Ip6::Address &             aAddress,
+                                         ThreadStatusTlv::DuaStatus aStatus,
+                                         const Child &              aChild)
+{
+    Coap::Message *  message = nullptr;
+    Ip6::MessageInfo messageInfo;
+    otError          error;
+
+    VerifyOrExit((message = Get<Coap::Coap>().NewPriorityMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(error =
+                      message->Init(OT_COAP_TYPE_CONFIRMABLE, OT_COAP_CODE_POST, OT_URI_PATH_DUA_REGISTRATION_NOTIFY));
+    SuccessOrExit(error = message->SetPayloadMarker());
+
+    SuccessOrExit(error = Tlv::AppendUint8Tlv(*message, ThreadTlv::kStatus, static_cast<uint8_t>(aStatus)));
+    SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &aAddress, sizeof(aAddress)));
+
+    messageInfo.GetPeerAddr().SetToRoutingLocator(Get<Mle::MleRouter>().GetMeshLocalPrefix(),
+                                                  HostSwap16(aChild.GetRloc16()));
+    messageInfo.SetPeerPort(kCoapUdpPort);
+    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
+
+    SuccessOrExit(error = Get<Coap::Coap>().SendMessage(*message, messageInfo));
+
+    otLogInfoDua("Sent ADDR_NTF for child %04x DUA %s", aChild.GetRloc16(), aAddress.ToString().AsCString());
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        // TODO: (DUA) (P4) may enhance to  guarantee the delivery of DUA.ntf
+        otLogWarnDua("Sent ADDR_NTF for child %04x DUA %s Error %s", aChild.GetRloc16(),
+                     aAddress.ToString().AsCString(), otThreadErrorToString(error));
+
+        if (message != NULL)
+        {
+            message->Free();
+        }
+    }
+}
+
+void DuaManager::UpdateChildDomainUnicastAddress(const Child &aChild, Mle::ChildDuaState aState)
+{
+    uint16_t childIndex = Get<ChildTable>().GetChildIndex(aChild);
+
+    if (aState == Mle::kChildDuaRemoved || aState == Mle::kChildDuaChanged)
```

Comment:
```suggestion
    if (aState == Mle::kChildDuaRemoved)
```
Checking for `kChildDuaChanged` is intended?

Reply:
yes, for example, if the fixed interface identifier is specified and then cleared, one new DUA will be generated. I think we should support the detection of this scenario and register the new one.